composer/utils/checkpoint.py

```diff
@@ -752,40 +750,6 @@ def filter_func(state_dict: dict) -> None:
     return filter_func
 
 
-def _validate_save_planner(save_planner: Optional[Any]) -> None:
```

Comment:
mind adding a bit to the PR description about what save planner and load planner do / why torch needs them when loading sharded checkpoints? That would help the rest of the team get more familiar with checkpointing code